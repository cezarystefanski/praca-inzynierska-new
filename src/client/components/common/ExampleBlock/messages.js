export default {
  BlockExplanation: {
    id: 'common--example_block--block_explanation',
  },
};
