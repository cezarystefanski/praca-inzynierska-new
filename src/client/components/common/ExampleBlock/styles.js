import { withStyles } from '../../../styles/themes/ExampleTheme/ExampleTheme';

export default withStyles(({ color, font }) => ({
  container: {
    backgroundColor: color.icyBlue,
    padding: '25px 50px',
    marginTop: '30px',
    borderRadius: '15px',
  },
  heading: {
    fontSize: font.sizes.big,
    fontWeight: font.weights.bold,
  },
  optionalParagraph: {
    fontSize: font.sizes.normal,
  },
}));
