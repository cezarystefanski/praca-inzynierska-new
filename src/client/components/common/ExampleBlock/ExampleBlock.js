import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { css } from 'styles/themes/ExampleTheme/ExampleTheme';

import messages from './messages';
import withStyles from './styles';

const ExampleBlockView = ({ text, styles }) => (
  <div {...css(styles.container)}>
    <p {...css(styles.heading)}><FormattedMessage {...messages.BlockExplanation} /></p>
    {text && <p {...css(styles.optionalParagraph)}>{text}</p>}
  </div>
);

ExampleBlockView.propTypes = {
  text: PropTypes.string,
};

ExampleBlockView.defaultProps = {
  text: '',
};

export const ExampleBlock = withStyles(ExampleBlockView);
