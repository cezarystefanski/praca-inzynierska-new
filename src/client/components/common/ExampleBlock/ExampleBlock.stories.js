import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs/react';

import { ExampleBlock } from './ExampleBlock';

const stories = storiesOf('common/ExampleBlock', module);
stories.addDecorator(withKnobs);

stories
  .add('without props', () => (
    <ExampleBlock />
  ))
  .add('with some props', () => (
    <ExampleBlock text="some example text" />
  ))
  .add('with dynamic variables', () => {
    const textProp = text('Example text', 'this is an example text');

    return <ExampleBlock text={textProp} />;
  });
