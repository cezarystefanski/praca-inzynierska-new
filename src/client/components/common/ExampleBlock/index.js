import { compose } from 'recompose';

import { ExampleBlockView as ExampleBlock } from './ExampleBlock';

import withStyles from './styles';

const enhance = compose(
  withStyles,
);

const Component = enhance(ExampleBlock);

export default Component;
