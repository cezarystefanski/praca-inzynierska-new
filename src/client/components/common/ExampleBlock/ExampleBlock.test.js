import React from 'react';
import { shallow } from 'enzyme';

import { ExampleBlock } from './ExampleBlock';

describe('component common/ExampleBlock', () => {
  const wrapper = shallow(<ExampleBlock text="example text" />);

  it('should render correctly with a prop', () => {
    expect(wrapper.find('p'))
      .toHaveLength(2);
  });

  it('should not differ from earlier version', () => {
    expect(wrapper)
      .toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const wrapperWithNoProps = shallow(<ExampleBlock />);

    expect(wrapperWithNoProps.find('p'))
      .toHaveLength(1);
  });
});
