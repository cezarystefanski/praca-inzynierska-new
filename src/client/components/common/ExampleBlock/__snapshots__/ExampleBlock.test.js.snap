// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component common/ExampleBlock should not differ from earlier version 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExampleBlockView
    css={[Function]}
    styles={
      Object {
        "container": Object {
          "_definition": Object {
            "backgroundColor": "#ccf",
            "borderRadius": "15px",
            "marginTop": "30px",
            "padding": "25px 50px",
          },
          "_len": 89,
          "_name": "container_1sms46k",
        },
        "heading": Object {
          "_definition": Object {
            "fontSize": "20px",
            "fontWeight": 700,
          },
          "_len": 36,
          "_name": "heading_1i3o6v",
        },
        "optionalParagraph": Object {
          "_definition": Object {
            "fontSize": "14px",
          },
          "_len": 19,
          "_name": "optionalParagraph_1b2af1e",
        },
      }
    }
    text="example text"
    theme={
      Object {
        "color": Object {
          "grey": "#aaa",
          "icyBlue": "#ccf",
          "white": "#fff",
        },
        "font": Object {
          "sizes": Object {
            "big": "20px",
            "normal": "14px",
            "small": "12px",
          },
          "weights": Object {
            "bold": 700,
            "normal": 500,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="heading_1i3o6v"
        >
          <FormattedMessage
            id="common--example_block--block_explanation"
            values={Object {}}
          />
        </p>,
        <p
          className="optionalParagraph_1b2af1e"
        >
          example text
        </p>,
      ],
      "className": "container_1sms46k",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FormattedMessage
            id="common--example_block--block_explanation"
            values={Object {}}
          />,
          "className": "heading_1i3o6v",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "common--example_block--block_explanation",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "example text",
          "className": "optionalParagraph_1b2af1e",
        },
        "ref": null,
        "rendered": "example text",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="heading_1i3o6v"
          >
            <FormattedMessage
              id="common--example_block--block_explanation"
              values={Object {}}
            />
          </p>,
          <p
            className="optionalParagraph_1b2af1e"
          >
            example text
          </p>,
        ],
        "className": "container_1sms46k",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormattedMessage
              id="common--example_block--block_explanation"
              values={Object {}}
            />,
            "className": "heading_1i3o6v",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "common--example_block--block_explanation",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "example text",
            "className": "optionalParagraph_1b2af1e",
          },
          "ref": null,
          "rendered": "example text",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
